ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/sdpa/kernels/dataflow/writer_decode_all.cpp

```diff
@@ -54,31 +55,46 @@ void fill_tile_partial(uint32_t cb_id, uint32_t tile_id, uint32_t cur_pos_in_til
         return;
     }
     // DPRINT << "Fill partial tile" << ENDL();
-    const uint16_t scalar_val = partial_val>>16;
-    volatile tt_l1_ptr uint16_t* ptr = reinterpret_cast<volatile tt_l1_ptr uint16_t*>(get_write_ptr(cb_id) + tile_id*tile_bytes);
-    int phase_start = (cur_pos_in_tile < 15) ? 0:1;
-    uint32_t fill_pos_in_phase = (cur_pos_in_tile+1) % 16;
-    // DPRINT << "phase_start: " << phase_start << ENDL();
-    // DPRINT << "fill_pos_in_phase: " << fill_pos_in_phase << ENDL();
-    if (phase_start == 0) {
-        // DPRINT << "Fill second and fourth phase" << ENDL();
+    const uint16_t datum_val = partial_val>>16;
+    volatile tt_l1_ptr uint16_t* uint16_ptr = reinterpret_cast<volatile tt_l1_ptr uint16_t*>(get_write_ptr(cb_id) + tile_id*tile_bytes);
+    volatile tt_l1_ptr uint32_t* uint32_ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_write_ptr(cb_id) + tile_id*tile_bytes);
+    int face_start = (cur_pos_in_tile < 15) ? 0:1;
+    uint32_t fill_pos_in_face = (cur_pos_in_tile+1) % 16;
+    // DPRINT << "face_start: " << face_start << ENDL();
+    // DPRINT << "fill_pos_in_face: " << fill_pos_in_face << ENDL();
+    if (face_start == 0) {
+        // DPRINT << "Fill second and fourth face" << ENDL();
+        // Fill 2 datums in each writes to optimize for performance
+        constexpr int num_uint32_datums_tile_face = 128;  // (16 * 16) / 2
```

Comment:
I think it would be simpler to use (16 * 16) / 2 in this expression. It will be evaluated at compile time, so there is no performance impact.

Reply:
Yes updated